src/composables/websocket.ts

```diff
@@ -0,0 +1,172 @@
+    try {
+      const data = camelcaseKeysDeep(JSON.parse(message.data));
+      if (
+        !data.subscription
+        || !subscribers[data.subscription as WebSocketChannelTypes]
+      ) {
+        return;
+      }
+      // Call all subscribers for the channel
+      Object.values(subscribers[data.subscription as WebSocketChannelTypes]).forEach(
+        (subscriberCb: any) => subscriberCb(data.payload),
+      );
+    } catch (error) {
+      handleUnknownError(error);
+    }
+  }
+
+  /**
+   * Subscribes to websocket channels.
+   */
+  function subscribeToWebSocketChannels() {
+    // Add an event listener for messages
+    wsClient.addEventListener('message', handleWebsocketMessage);
+
+    // Subscribe to all the channels
+    (Object.keys(channels) as WebSocketChannelTypes[])
+      .forEach(subscribeToWebSocketChannel);
+  }
+
+  /**
+   * UnSubscribes websocket channels.
+   */
+  function unsubscribeFromWebSocketChannels() {
+    // Remove event listener for messages
+    wsClient.removeEventListener('message', handleWebsocketMessage);
+
+    // UnSubscribe all channels
+    Object.keys(channels).forEach((channelName: string) => {
+      channels[channelName as WebSocketChannelTypes] = false;
+    });
+  }
+
+  /**
+   * This function creates a new WebSocket client and attempts to connect to the specified server.
+   * @param {string} url - The URL of the WebSocket server to connect to
+   */
+  async function connectToWebSocketServer(url: string) {
+    if (wsClient) {
+      unsubscribeFromWebSocketChannels();
+
+      if (isWsConnected.value) {
+        wsClient.close();
+      }
+    }
+
+    isWsConnected.value = false;
+
+    wsClient = new WebSocket(url);
+
+    wsClient.onopen = () => {
+      isWsConnected.value = true;
+      subscribeToWebSocketChannels();
+    };
+
+    wsClient.onclose = () => {
+      isWsConnected.value = false;
+    };
+  }
+
+  /**
+   * Subscribes to a specific websocket channel
+   * @param {WebSocketChannelTypes} channel - The websocket channel to subscribe to.
+   * @param {Function} [callback=(payload: any) => payload]
+   *    - The function to be called when a message is received on the specified channel.
+   * @returns {Function}
+   *    - A function that can be called to unsubscribe the specified callback from the channel.
+   */
+  function watchChannelUpdates(
+    channel: WebSocketChannelTypes,
+    // eslint-disable-next-line no-unused-vars
+    callback: (payload: any) => void,
+  ): () => void {
+    // subscribe to channel
+    subscribeToWebSocketChannel(channel);
+
+    // register the subscriber
+    if (!subscribers[channel]) {
+      subscribers[channel] = [];
+    }
+
+    const subscriberId = genUuid();
+    subscribers[channel][subscriberId] = callback;
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      delete subscribers[channel][subscriberId];
+    };
+  }
+
+  watch(
+    () => activeNetwork.value,
+    (_activeNetwork: INetwork) => {
```

Comment:
It's rather subjective, but I suggest to not use variable names prefixed by underscore as it means it's a temporary thing. If there's a name collision you can use `val` here.